pkg/util/sds.go

```diff
@@ -1,110 +1,205 @@
-package util
-import "fmt"
-import "strings"
-var(
+package pkg
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+const SDS_MAX_PREALLOC int = 1024 * 1024
+
+var (
 	str string
-	i int = 0
-	fr int = 10
-	bu[] byte
+	i   int = 0
+	fr  int = 0
 
-	choose string = ""
+	bu         []byte
+	sdslen     int
+	choose     string = ""
+	cat_or_not int    = 0
 )
+
 type sdshdr struct {
-// 等于SDS所保存字符串的长度
+	// 等于SDS所保存字符串的长度
 
 	len int
- // 记录buf数组中未使用字节的数量
+	// 记录buf数组中未使用字节的数量
 	free int
- // 字节数组，用于保存字符串
-	buf[] byte
- }
-
+	// 字节数组，用于保存字符串
+	buf []byte
+}
 
- //TODO:makeroom
- func sdsMakeRoomFor(s sdshdr){
+//TODO:makeroom
+func sdsMakeRoomFor(s *sdshdr) {
+	if s.len > SDS_MAX_PREALLOC {
+		fr *= 2
+	} else {
+		fr += SDS_MAX_PREALLOC
+	}
+}
 
- }
+func SetFunc(key string, value string) {
 
- func SetFunc(key string, value string){
 	sdsNew(value)
 
- }
-
- func GetFunc(key string){
+}
 
- }
+func GetFunc(key string) {
 
+}
 
- /*func split(s *sdshdr) string{
+/*func split(s *sdshdr) string{
 	s.buf = []byte(str)
 	for i, value:=range s.buf{
 	fmt.Printf("buf%d:'%c'",i,value)
 	}
-	return "OK" 
+	return "OK"
  }*/
 
- func UserAsk(ask_str string){
+func UserAsk(ask_str string) {
 	ask_split := strings.Split(ask_str, " ")
-	if ask_split[0] == "set" && ask_split[0] == "SET"{
+	if ask_split[0] == "set" && ask_split[0] == "SET" {
```

Comment:
I think you should change it to this style:
```go
if x := ask_split[0]; x == "set" && x == "SET" {
```
In addition, you should reconsider the operator `&&`. Does it make sense?